components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -1424,6 +1426,25 @@ public static String getIDTokenIssuer() {
         return issuer;
     }
 
+    public static boolean isOrganizationValidAndActive(String organizationId) throws IdentityOAuth2Exception {
+
+        try {
+            boolean organizationExistById = OAuth2ServiceComponentHolder.getInstance().getOrganizationManager()
+                    .isOrganizationExistById(organizationId);
+            if (!organizationExistById) {
+                return false;
+            }
```

Comment:
As the authorized organization is set by the server side, it can't have invalid value. But there is a posibility the org has been deleted while processing this logic. In such case will it throw error or null. If doesn't throw an error, we can reduce above check.

Reply:
"isOrganizationExistById" returns false if the org id is invalid/ org is deleted

Is your concern is org get deleted at a point when isOrganizationExistById check is completed and it returned true, but at the point of getOrganization method is invoked the org has been deleted ?

Reply:
No, Tried to remove the first method call, if the 2nd method call can be used to get org and check org exist

Reply:
This is fine.